content/blog/2025-06-13-introducing-valkey-search/index.md

```diff
@@ -0,0 +1,173 @@
+
+
+## Performance & Low Latency
+
+Valkey-search is designed as an in-memory secondary index, achieving exceptional performance. A multi-threaded architecture optimizes query and mutation processing with minimal thread contention, enabling near-linear vertical scalability.
+
+At its core, valkey-search’s threading architecture follows a common design pattern: a worker thread pool combined with task queues. It employs advanced synchronization mechanisms to maintain index consistency while minimizing contention among worker threads. By time-slicing CPU access between read and write operations, the system enables an almost lock-free read path, delivering high performance and consistently low search latency. 
+
+Valley-search’s HNSW implementation is based on the OSS project [HNSWLib](https://github.com/nmslib/hnswlib). While HNSWLib is well-regarded for its speed, we have enhanced its performance, resiliency and its production maturity. These improvements include better `SIMD` utilization, promotion of CPU cache efficiency, memory utilization and more.
+
+
+## Future Enhancements
+
+This first release of valkey-search focuses on vector search, but it is designed as a general-purpose indexing subsystem. Future releases will extend both the available data types as well as provide post-query data processing facilities.
+
+
+## Get Involved
+
+Valkey-search is open source and ready for you to explore. Whether you're building cutting-edge AI applications or integrating vector search into existing systems, we invite you to try it out. The easiest way to get started is by visiting the [GitHub repository](https://github.com/valkey-io/valkey-search), where you'll find setup instructions, documentation, and examples.
+
+We welcome contributions of all kinds - code, documentation, testing, and feedback. Join the community, file issues, open pull requests, or suggest improvements. Your involvement helps make valkey-search better for everyone.
+
+Ready to dive in? Clone the repo, fire up the dev container, and start building high-performance vector search with valkey-search.
+
+
+## About the Authors
+
+
+<table>
+  <tr>
+   <td>
+     
+![Yair Gottdenker](images/yair.png)
+
+   </td>
+   <td>
+
+### Yair Gottdenker
+
+Yair Gottdenker is a Senior Staff Software Engineer at Memorystore GCP. 
+   </td>
+  </tr>
+  <tr>
+   <td>
+     
+![Allen Samuels](images/Allen.png)
+
+   </td>
+   <td>
+     
+### Allen Samuels 
+ 
+Allen Samuels is a Principal Engineer at AWS.
+   </td>
+  </tr>
+</table>

```

Comment:
This is part of the metadata that should be embedded. See https://github.com/valkey-io/valkey-io.github.io/blob/2b956fa82d018c0f96d2cb384dddce50543a6fb5/content/blog/2024-08-29-valkey-memory-efficiency-8-0.md?plain=1#L1 for how it's done.

For a full example you can look at https://github.com/valkey-io/valkey-io.github.io/commit/837e250c9f361222e16060771da1a5a4a410260b